lib/autocomplete-ruby-client.coffee

```diff
@@ -1,61 +1,105 @@
-    # This means that running two atoms and closing one, kills rsense for the other
-    return if @rsenseStarted == true
-    @rsenseStarted = true
+    exec("ps -ef | head -1; ps -ef | grep java",
+      (error, stdout, stderr) ->
+        if error != null
+          atom.notifications.addError('Error looking for resense process',
+              {detail: "exec error: #{error}", dismissable: true}
+            )
+        else
+          @rsenseProcess = $.grep(TableParser.parse(stdout), (process) ->
+            process.CMD.join(' ').match( /rsense.*--port.*--path/ )
+          )[0]
+          if @rsenseProcess == undefined || @rsenseProcess == null
+            start()
+          else
+            @rsenseStarted = true
+    )
+
+  # Before trying to start in Windows we need to kill any existing rsense servers, so
+  # as to not end up with multiple rsense servsers unkillable by 'rsense stop'
+  # This means that running two atoms and closing one, kills rsense for the other
+  startRsenseWin32: =>
+    return if @rsenseStarted
+    start = @startRsenseCommand
 
     exec("#{@rsensePath} stop",
       (error, stdout, stderr) =>
         if error == null
-
-          port = atom.config.get('autocomplete-ruby.port')
-          exec("#{@rsensePath} start --port #{port} --path #{@projectPath}",
-            (error, stdout, stderr) ->
-              if error != null
-                atom.notifications.addError('Error starting rsense',
-                    {detail: "exec error: #{error}", dismissable: true}
-                  )
-                @rsenseStarted = false
-          )
-
+          start()
         else
           atom.notifications.addError('Error stopping rsense',
               {detail: "exec error: #{error}", dismissable: true}
             )
           @rsenseStarted = false
     )
 
-  stopRsense: ->
+  startRsenseCommand: =>
+    return if @rsenseStarted
+    exec("#{@rsensePath} start --port #{@port} --path #{@projectPath}",
+      (error, stdout, stderr) ->
+        if error != null
+          atom.notifications.addError('Error starting rsense',
+              {detail: "exec error: #{error}", dismissable: true}
+            )
+        else
+          @rsenseStarted = true
+    )
+
+  # First count how many atom windows are open.
+  # If there is only one open, then kill the rsense process.
+  # This is also able to kill an rsense process without a pid file.
+  # Otherwise do nothing so you will still be able to use rsense in other windows.
+  stopRsenseUnix: =>
+    stopCommand = @stopRsense
+
+    exec("ps -ef | head -1; ps -ef | grep atom",
+      (error, stdout, stderr) ->
+        if error != null
+          atom.notifications.addError('Error looking for atom process',
+              {detail: "exec error: #{error}", dismissable: true}
+            )
+        else
+          @atomProcesses = $.grep(TableParser.parse(stdout), (process) ->
+            process.CMD.join(' ').match( /--type=renderer.*--node-integration=true/ )
```

Comment:
Yes. Electron, on which Atom is built, has a main process which launches the renderers. Those are basically the windows that display everything. http://electron.atom.io/docs/tutorial/quick-start/

So `--node-integration=true` indicates wether the window has node running. This option was also there since the beginning. I added it because once you open the dev tools it also has it's own renderer process. But the difference is that it's not running node like the real editor windows. Just a way to distinguish between the dev tools and real windows.

About testing it in Windows:
I tried to test it with Windows 10 but rsense won't even work in Windows for me. When I try to run `rsense start` I get:

`uninitialized constant Spoon::FileActions (NameError)`

So can't really test it then with Windows unfortunately.